tests/test_tags.py

```diff
@@ -285,26 +285,31 @@ def test_mac_platforms(self):
 
 
 class TestManylinuxPlatform:
-    def test_module_declaration_true(self, manylinux_module):
-        manylinux_module.manylinux1_compatible = True
-        assert tags._is_manylinux_compatible("manylinux1", (2, 5))
+    def teardown_method(self):
+        # Clear the version cache
+        tags._glibc_version = []
 
-    def test_module_declaration_false(self, manylinux_module):
-        manylinux_module.manylinux1_compatible = False
-        assert not tags._is_manylinux_compatible("manylinux1", (2, 5))
-
-    def test_module_declaration_missing_attribute(self, manylinux_module):
-        try:
-            del manylinux_module.manylinux1_compatible
-        except AttributeError:
-            pass
-        assert not tags._is_manylinux_compatible("manylinux1", (2, 5))
+    @pytest.mark.parametrize("tf", (True, False))
+    @pytest.mark.parametrize(
+        "attribute,glibc", (("1", (2, 5)), ("2010", (2, 12)), ("2014", (2, 17)))
+    )
+    def test_module_declaration(self, manylinux_module, attribute, glibc, tf):
+        manylinux = "manylinux{}_compatible".format(attribute)
+        setattr(manylinux_module, manylinux, tf)
+        res = tags._is_manylinux_compatible(manylinux, "x86_64", glibc)
+        assert tf is res
+        delattr(manylinux_module, manylinux)
```

Comment:
It's easiest to use pytest's `monkeypatch` for temporarily setting an attribute:

```suggestion
    def test_module_declaration(self, monkeypatch, manylinux_module, attribute, glibc, tf):
        manylinux = "manylinux{}_compatible".format(attribute)
        monkeypatch.setattr(manylinux_module, "manylinux", tf)
        res = tags._is_manylinux_compatible(manylinux, "x86_64", glibc)
        assert tf is res
```

Reply:
I tried that. `monkypatch` cannot use `setattr` on an attribute that was not already present, I get
```
>       monkeypatch.setattr(manylinux_module, manylinux, tf)
E       AttributeError: <module '_manylinux'> has no attribute \
                     'manylinux2014_compatible'
```

Reply:
Try [using `raising=False`](https://docs.pytest.org/en/latest/monkeypatch.html).